File: Obligatorio1/Obligatorio1.Domain/Purchase.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Obligatorio1.Domain
{
    public class Purchase
    {
        public int PurchaseID { get; set; }
        public int UserID { get; set; }
        public string UserName { get; set; }

        public string EmailUsuario { get; set; }
        public User User { get; set; }
        public List<PurchaseProduct> PurchasedProducts { get; set; }

        public string PromoApplied { get; set; }
        
        [Column(TypeName = "datetime2")]
        public DateTime? DateOfPurchase { get; set; }

        public string PaymentMethod { get; set; }

        public Purchase()
        {
            this.PurchasedProducts = new List<PurchaseProduct>();
            this.PromoApplied = String.Empty;
            this.DateOfPurchase = null;
            this.PaymentMethod = String.Empty;
            this.UserName = String.Empty;
            this.EmailUsuario = String.Empty;
        }

        public Purchase(int PurchaseID, int userID, List<PurchaseProduct> PurchasedProducts, string NamePromoApplied, DateTime DateOfPurchase, 
                        string UserName, string EmailUsuario, string PaymentMethod)
        {
            this.PurchaseID = PurchaseID;
            this.UserID = userID;
            this.PurchasedProducts = PurchasedProducts;
            this.PromoApplied = NamePromoApplied;
            this.DateOfPurchase = DateOfPurchase;
            this.PaymentMethod = PaymentMethod;
            this.UserName= UserName;
            this.EmailUsuario = EmailUsuario;
        }
    }
}

File: Obligatorio1Angular/src/app/modelos/Purchase.ts
import { PurchaseProduct } from "./PurchaseProduct";
import { User } from "./User";

export class Purchase {
    purchaseID: number;
    userID: number;
    user: User;
    userName: string;
    purchasedProducts: PurchaseProduct[];
    promoApplied: string;
    dateOfPurchase: Date | null;
    paymentMethod: string;
    emailUsuario: string;

    constructor(purchaseID:number,userID:number,user:User,purchasedProducts:PurchaseProduct[],promoApplied:string,dateOfPurchase:Date,paymentMethod:string, 
                userName:string, emailUsuario:string){
        this.purchaseID=purchaseID;
        this.userID=userID;
        this.user=user;
        this.purchasedProducts=purchasedProducts;
        this.promoApplied=promoApplied;
        this.dateOfPurchase=dateOfPurchase;
        this.paymentMethod=paymentMethod;
        this.userName= userName;
        this.emailUsuario= emailUsuario;
    }
}